crates/gateway-types/src/websocket.rs

```diff
@@ -0,0 +1,35 @@
+// Types used for web socket subscription events
+use crate::reply::Block;
+use serde::Serialize;
+use tokio::sync::broadcast;
+
+#[derive(Debug, Clone, Serialize)]
+pub struct WebsocketEventSync {
+    pub block: Box<Block>,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct WebsocketEventNewHead {
+    pub block: Box<Block>,
```

Comment:
Is `Box` the correct container type to use? How does jsonrpsee share this data between different subscriptions?

i.e. if we have `N` new head subscriptions, will it copy `Box<T>` `N` times, or does it somehow magically re-use the single one? If it is copied `N` times, then we likely want this to be `Rc<Block>` or `Arc<Block>` instead.

Reply:
`broadcast::Sender` source code seemed to have something like an `Arc` built in.
Also did some profiling,

```
Broadcasting struct 10000 listeners: 5260224 bytes
Broadcasting arc 10000 listeners:    5227392 bytes
Broadcasting num u8 10000 listeners: 5129088 bytes
Duplicating struct 10000 times:      9421632 bytes
Duplicating 10000x with 1s hold:    13206464 bytes
```

Details here,
https://github.com/matthieuauger/pathfinder/issues/7#issuecomment-1471819271

Reply:
I had a look into the tokio source code and it appears they use an internal `RwLock` queue which the `T` gets copied into for sharing.

Now I'm wondering if we even need the `Box` but its probably fine as is.